### PR TITLE
Gpl 769 2

### PR DESCRIPTION
**Pull request by harrietc52**
Checked out from GPL-857

### CODE REVIEW

**Review by stevieing — APPROVED** [review]

File: migrations/helpers/shared_helper.py [hunk 1]
```diff
@@ -131,78 +116,94 @@ def get_cherrypicked_samples(
...
-        f" JOIN {events_wh_db}.subjects AS mlwh_events_subjects ON (mlwh_events_subjects.uuid = UNHEX(REPLACE(mlwh_lh_sample.lh_sample_uuid, '-', '')))"  # noqa: E501
-        f" JOIN {events_wh_db}.roles AS mlwh_events_roles ON (mlwh_events_roles.subject_id = mlwh_events_subjects.id)"  # noqa: E501
-        f" JOIN {events_wh_db}.events AS mlwh_events_events ON (mlwh_events_events.id = mlwh_events_roles.event_id)"  # noqa: E501
-        f" JOIN {events_wh_db}.event_types AS mlwh_events_event_types ON (mlwh_events_event_types.id = mlwh_events_events.event_type_id)"  # noqa: E501
-        f" WHERE mlwh_sample.description IN %(root_sample_ids)s"
-        f" AND mlwh_lh_sample.plate_barcode IN %(plate_barcodes)s"
-        f" AND mlwh_events_event_types.key = '{PLATE_EVENT_DESTINATION_CREATED}'"
-        " GROUP BY mlwh_sample.description, mlwh_lh_sample.plate_barcode, mlwh_sample.phenotype, mlwh_lh_sample.coordinate;"  # noqa: E501
+    cherry_picked_sets = [{cp_sample[0], cp_sample[1]} for cp_sample in cherry_picked_samples]
+    return list(
```
Comment (pjvv) [inline migrations/helpers/shared_helper.py:204]: I think first storing the filter as a variable then returning the list will be more readable.

File: setup_test_db.py [hunk 2]
```diff
@@ -333,6 +333,31 @@
 ) ENGINE=InnoDB AUTO_INCREMENT=17 DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
 """
 
+create_cherrypicked_samples_view = """
```
Comment (pjvv) [inline setup_test_db.py:336]: It's not great to store the view's definition in every place we need it to test, any chance to of using the container or is that too large a change? Worth another story?

File: tests/migrations/helpers/test_shared_helper.py [hunk 3]
```diff
@@ -290,21 +299,43 @@ def test_get_cherrypicked_samples_chunking_sentinel_and_beckman(config):
...
-        pd.DataFrame(["MCM003"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0]),  # Sentinel query response
-        pd.DataFrame(["MCM003", "MCM004"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0, 1]),  # Beckman query response
-        pd.DataFrame(["MCM005"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0]),  # Sentinel query response
-        pd.DataFrame(["MCM005", "MCM006"], columns=[FIELD_ROOT_SAMPLE_ID], index=[0, 1]),  # Beckman query response
+        pd.DataFrame(
+            [
+                "MCM001",
+                "MCM001",
+                "MCM002",
+            ],  # Sentinel  # Beckman  # Beckman
```
Comment (pjvv) [inline tests/migrations/helpers/test_shared_helper.py:309]: Double inline comments here.

File: tests/migrations/helpers/test_shared_helper.py [hunk 4]
```diff
@@ -290,21 +299,43 @@ def test_get_cherrypicked_samples_chunking_sentinel_and_beckman(config):
...
+            columns=[FIELD_ROOT_SAMPLE_ID],
+            index=[0, 1, 2],
+        ),
+        pd.DataFrame(
+            [
+                "MCM003",
+                "MCM003",
+                "MCM004",
+            ],  # Sentinel  # Beckman  # Beckman
```
Comment (pjvv) [inline tests/migrations/helpers/test_shared_helper.py:318]: See previous.

File: tests/migrations/helpers/test_shared_helper.py [hunk 5]
```diff
@@ -290,21 +299,43 @@ def test_get_cherrypicked_samples_chunking_sentinel_and_beckman(config):
...
+            columns=[FIELD_ROOT_SAMPLE_ID],
+            index=[0, 1, 2],
+        ),
+        pd.DataFrame(
+            [
+                "MCM005",
+                "MCM005",
+                "MCM006",
+            ],  # Sentinel  # Beckman  # Beckman
```
Comment (pjvv) [inline tests/migrations/helpers/test_shared_helper.py:327]: See previous.

**Review by pjvv — CHANGES_REQUESTED** [review]

File: crawler/sql_queries.py [hunk 6]
```diff
@@ -120,3 +123,20 @@
 SQL_DART_SET_WELL_PROPERTY = "{CALL dbo.plDART_PlateUpdateWell (?,?,?,?)}"
 
 SQL_DART_ADD_PLATE = "{CALL dbo.plDART_PlateCreate (?,?,?)}"
+
+SQL_MLWH_GET_CP_SAMPLES = f"""
```
[the remark this answers is not in the document]
Comment (harrietc52) [reply crawler/sql_queries.py:127]: Made these changes in GPL-769-refactor-cherrypicked-helper

File: crawler/sql_queries.py [hunk 7]
```diff
@@ -120,3 +123,20 @@
...
+    SELECT root_sample_id AS `{FIELD_ROOT_SAMPLE_ID}`, `{FIELD_PLATE_BARCODE}`,
+     phenotype AS `Result_lower`, `{FIELD_COORDINATE}`
+    FROM cherrypicked_samples
+    WHERE root_sample_id IN %(root_sample_ids)s
+    AND `{FIELD_PLATE_BARCODE}` IN %(plate_barcodes)s
+"""
+
+SQL_MLWH_GET_CP_SAMPLES_BY_DATE = f"""
```
[the remark this answers is not in the document]
Comment (harrietc52) [reply crawler/sql_queries.py:135]: Made these changes in GPL-769-refactor-cherrypicked-helper

File: setup_test_db.py (hunk 2 above)
Comment (harrietc52) [reply setup_test_db.py:336]: We could put it down as a new story but if there is going to be a new cherrypick service, it could be considered then?

**Comment by harrietc52** [comment]
https://github.com/sanger/crawler/pull/292 was checkouted from this branch, so we will use that refactor branch to deploy to UAT as it contains all these changes plus more

**Pull request merged**